Kudu.Core/Scan/ScanManager.cs

```diff
@@ -0,0 +1,442 @@
+            });
+
+                return obj;
+        }
+
+        public async Task<ScanReport> GetScanResultFile(String scanId,String mainScanDirPath)
+        {
+            //JObject statusRes = await GetScanStatus(scanId, mainScanDirPath);
+            ScanReport report = null;
+            //Run task to read the result file
+            await Task.Run(() =>
+            {
+                String report_path = Path.Combine(mainScanDirPath, Constants.ScanFolderName + scanId, Constants.ScanLogFile);
+                ScanStatusResult scr = ReadScanStatusFile(scanId, mainScanDirPath, Constants.ScanStatusFile, null);
+                report = new ScanReport();
+                report.Id = scr.Id;
+                report.Timestamp = DateTime.ParseExact(scr.Id,DATE_TIME_FORMAT, System.Globalization.CultureInfo.InvariantCulture).ToUniversalTime();
+                String text = "Report file not yet generated. The scan might be still running, please check the status";
+
+                if (FileSystemHelpers.FileExists(report_path))
+                {
+                    text = FileSystemHelpers.ReadAllTextFromFile(report_path);
+                }
+
+                report.Report = text;
+                
+            });
+
+            //All the contents of the file and the timestamp
+                return report;
+        }
+
+        private static ScanStatusResult ReadScanStatusFile(String scanId, String mainScanDirPath, String fileName,String folderName)
+        {
+            ScanStatusResult obj = null;
+            String readPath = Path.Combine(mainScanDirPath, Constants.ScanFolderName + scanId, fileName);
+
+            //Give preference to folderName if given
+            if(folderName != null)
+            {
+                readPath = Path.Combine(folderName, fileName);
+            }
+
+            //Check if scan status file has been formed
+            if (FileSystemHelpers.FileExists(readPath))
+            {
+                //Read json file and deserialize into JObject
+                using (FileStream file = System.IO.File.OpenRead(readPath))
+                {
+                    using (StreamReader sr = new StreamReader(file))
+                    {
+                        JsonSerializer serializer = new JsonSerializer();
+                        obj = (ScanStatusResult)serializer.Deserialize(sr, typeof(ScanStatusResult));
+                    }
+
+                }
+            }
+           
+            return obj;
+        }
+
+        public static async Task<bool> PerformBackgroundScan(ITracer _tracer, IOperationLock _scanLock, String folderPath,ExternalCommandFactory _externalCommandFactory,IEnvironment _environment,CancellationToken token)
+        {
+
+            var successfulScan = true;
+
+            await Task.Run(() =>
+            {
+
+                _scanLock.LockOperation(() =>
+                {
+
+                    String filePath = Path.Combine(folderPath, Constants.ScanStatusFile);
```

Comment:
I have exposed the endpoints through kudu for this, so the end point is /api/scan/stop. Since there is only one scan running at a time on a webapp, I believe we do not need the scanId. Basically I am creating a temp file when a scan starts, and deleting that file if stop is requested. The process will check if the file exists, if not, the scan is killed.

Reply:
I think it would still be good to add a unique id to this path